src/librustdoc/doctest.rs

```diff
@@ -291,7 +291,7 @@ fn run_test(
     for debugging_option_str in &options.debugging_opts_strs {
         compiler.arg("-Z").arg(&debugging_option_str);
     }
-    if no_run && !compile_fail {
+    if (no_run || options.no_run) && !compile_fail {
```

Comment:
I think this should be set as `no_run` from the arguments rather than here, see where `run_test` is called.